src/main/java/ch/njol/skript/expressions/ExprNearestEntity.java

```diff
@@ -0,0 +1,128 @@
+import org.bukkit.event.Event;
+import org.eclipse.jdt.annotation.Nullable;
+
+import java.util.Arrays;
+
+@Name("Nearest Entity")
+@Description("Gets the entity nearest to a location or another entity.")
+@Examples({
+	"kill the nearest pig and cow relative to player",
+	"teleport player to the nearest cow relative to player",
+	"teleport player to the nearest entity relative to player",
+	"",
+	"on click:",
+	"\tkill nearest pig"
+})
+@Since("INSERT VERSION")
+public class ExprNearestEntity extends SimpleExpression<Entity> {
+
+	static {
+		Skript.registerExpression(ExprNearestEntity.class, Entity.class, ExpressionType.COMBINED,
+			"[the] nearest %*entitydatas% [[relative] to %entity/location%]",
+			"[the] %*entitydatas% nearest [to %entity/location%]");
+	}
+
+	@SuppressWarnings("NotNullFieldNotInitialized")
+	private EntityData<?>[] entityDatas;
+
+	@SuppressWarnings("NotNullFieldNotInitialized")
+	private Expression<?> relativeTo;
+
+	@Override
+	@SuppressWarnings("unchecked")
+	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, ParseResult parseResult) {
+		entityDatas = ((Literal<EntityData<?>>) exprs[0]).getArray();
+		if (entityDatas.length != Arrays.stream(entityDatas).distinct().count()) {
+			Skript.error("Entity list may not contain duplicate entities");
+			return false;
+		}
+		relativeTo = exprs[1];
+		return true;
+	}
+
+
+	@Nullable
+	private Entity getNearestEntity(EntityData<?> entityData, Location relativePoint, @Nullable Entity excludedEntity) {
+		Entity nearestEntity = null;
+		double nearestDistance = -1;
+		for (Entity entity : relativePoint.getWorld().getEntitiesByClass(entityData.getType())) {
+			if (entity != excludedEntity && entityData.isInstance(entity)) {
+				double distance = entity.getLocation().distance(relativePoint);
+				if (nearestEntity == null || distance < nearestDistance) {
+					nearestDistance = distance;
+					nearestEntity = entity;
+				}
+			}
+		}
+		return nearestEntity;
+	}
+
+	@Override
+	protected Entity[] get(Event e) {
+		Object relativeTo = this.relativeTo.getSingle(e);
+		if (relativeTo == null) {
+			return new Entity[0];
+		}
+		Entity[] nearestEntities = new Entity[entityDatas.length];
+		for (int i = 0; i < nearestEntities.length; i++) {
+			if (relativeTo instanceof Entity) {
+				nearestEntities[i] = getNearestEntity(entityDatas[i], ((Entity) relativeTo).getLocation(), (Entity) relativeTo);
+			} else {
+				nearestEntities[i] = getNearestEntity(entityDatas[i], (Location) relativeTo, null);
+			}
+		}
+		return nearestEntities;
+	}
+
+	@Override
+	public boolean isSingle() {
+		return entityDatas.length == 1;
+	}
+
+	@Override
+	public Class<? extends Entity> getReturnType() {
+		return entityDatas.length == 1 ? entityDatas[0].getType() : Entity.class;
+	}
+
+	@Override
+	public String toString(@Nullable Event e, boolean debug) {
+		return "nearest " + StringUtils.join(entityDatas) + " relative to " + relativeTo.toString(e, debug);
```

Comment:
relativeTo can be null.
```suggestion
	public String toString(@Nullable Event event, boolean debug) {
		if (relativeTo == null)
			return "nearest " + StringUtils.join(entityDatas);
		return "nearest " + StringUtils.join(entityDatas) + " relative to " + relativeTo.toString(event, debug);
```

Reply:
when? I don't think `relativeTo` can't be null here

Reply:
Not anymore